### ISSUE TITLE
handle dimensions/spacing != int

### ISSUE THREAD

**Issue by bebbi** [pieper/cvg#7]
this leads to non-integer dimensions and bad array pointers.
some ideas:
- increase extents to match spacing
- change spacing to match extents
- round up dimensions and add one smaller space at end of volume
- throw error
- ignore this issue and target only extent auto-calculation from expression bounding box
  have you got a pref? I like the last.

**Comment by pieper** [comment]
Makes sense - we should only allow setting combinations of options that lead to a well-formed grid.  Setting from the bounding box will be good, but there will also be some subtrees where the bounding box cannot be known until the sampling is defined (i.e. if they contain some kinds of nonlinear transforms).
